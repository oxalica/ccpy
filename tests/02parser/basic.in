a
1 + 2 + 3
1 + 2 * ~~3
a + 1 % ...
1,
2, 3 # This will not be concated to previous line
pass
1, (2,), (3, 4), (5)
f(1, (2), 3 + 4(), ())()
()
True
True, False + None
'123', "234"
'3\n\'', "\\\""
global a
nonlocal b, c, d
return (1,)[0]
'123'[1, 2]
raise Exception()
a = 1
a, b = b, a
a = b.c = d, e = f()[g()] = 1, a
del a
del b, c
del d[0]
def f(): pass
def g(x): return x
def h(*xs): return xs
def i(a, b=1+2, *xs): return
def j():
  pass
def k():
  1
  def l():
    return
  return l

1 if 2 else 3
1 if 2 else 3 if 4 else 5
1 and 2 or not 3 and not not 4 and 5

if 1: 2

if 3: 4
else: 5

if 6: 7
elif 8: 9

if 10: 11
elif 12: 13
elif 14:
  if 16:
    17
else:
  18
