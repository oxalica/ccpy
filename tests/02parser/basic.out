StmtExpr(ExprName("a"))
StmtExpr(ExprBinary("add", ExprBinary("add", ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))), ExprLiteral(LitInteger(3))))
StmtExpr(ExprBinary("add", ExprLiteral(LitInteger(1)), ExprBinary("mul", ExprLiteral(LitInteger(2)), ExprUnary("inv", ExprUnary("inv", ExprLiteral(LitInteger(3)))))))
StmtExpr(ExprBinary("add", ExprName("a"), ExprBinary("mod", ExprLiteral(LitInteger(1)), ExprLiteral(LitEllipse))))
StmtExpr(ExprTuple(ExprLiteral(LitInteger(1))))
StmtExpr(ExprTuple(ExprLiteral(LitInteger(2)), ExprLiteral(LitInteger(3))))
StmtPass
StmtExpr(ExprTuple(ExprLiteral(LitInteger(1)), ExprTuple(ExprLiteral(LitInteger(2))), ExprTuple(ExprLiteral(LitInteger(3)), ExprLiteral(LitInteger(4))), ExprLiteral(LitInteger(5))))
StmtExpr(ExprCall(ExprCall(ExprName("f"), [ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)), ExprBinary("add", ExprLiteral(LitInteger(3)), ExprCall(ExprLiteral(LitInteger(4)), [])), ExprTuple()]), []))
StmtExpr(ExprTuple())
StmtExpr(ExprLiteral(LitBool(true)))
StmtExpr(ExprTuple(ExprLiteral(LitBool(true)), ExprBinary("add", ExprLiteral(LitBool(false)), ExprLiteral(LitNone))))
StmtExpr(ExprTuple(ExprLiteral(LitStr("123")), ExprLiteral(LitStr("234"))))
StmtExpr(ExprTuple(ExprLiteral(LitStr("3\n'")), ExprLiteral(LitStr("\\\""))))
StmtGlobal("a")
StmtNonlocal("b", "c", "d")
StmtReturn(Some(ExprIndex(ExprTuple(ExprLiteral(LitInteger(1))), ExprLiteral(LitInteger(0)))))
StmtExpr(ExprIndex(ExprLiteral(LitStr("123")), ExprTuple(ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)))))
StmtRaise(ExprCall(ExprName("Exception"), []))
StmtAssign([PatName("a")], ExprLiteral(LitInteger(1)))
StmtAssign([PatTuple[PatName("a"), PatName("b")]], ExprTuple(ExprName("b"), ExprName("a")))
StmtAssign([PatName("a"), PatAttr(ExprName("b"), "c"), PatTuple[PatName("d"), PatName("e")], PatIndex(ExprCall(ExprName("f"), []), ExprCall(ExprName("g"), []))], ExprTuple(ExprLiteral(LitInteger(1)), ExprName("a")))
StmtDel(PatName("a"))
StmtDel(PatTuple[PatName("b"), PatName("c")])
StmtDel(PatIndex(ExprName("d"), ExprLiteral(LitInteger(0))))
StmtDef("f", [], None, [StmtPass])
StmtDef("g", [("x", None)], None, [StmtReturn(Some(ExprName("x")))])
StmtDef("h", [], Some("xs"), [StmtReturn(Some(ExprName("xs")))])
StmtDef("i", [("a", None), ("b", Some(ExprBinary("add", ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)))))], Some("xs"), [StmtReturn(None)])
StmtDef("j", [], None, [StmtPass])
StmtDef("k", [], None, [StmtExpr(ExprLiteral(LitInteger(1))), StmtDef("l", [], None, [StmtReturn(None)]), StmtReturn(Some(ExprName("l")))])
StmtExpr(ExprCond(ExprLiteral(LitInteger(2)), ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(3))))
StmtExpr(ExprCond(ExprLiteral(LitInteger(2)), ExprLiteral(LitInteger(1)), ExprCond(ExprLiteral(LitInteger(4)), ExprLiteral(LitInteger(3)), ExprLiteral(LitInteger(5)))))
StmtExpr(ExprBinary("or", ExprBinary("and", ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))), ExprBinary("and", ExprUnary("not", ExprLiteral(LitInteger(3))), ExprBinary("and", ExprUnary("not", ExprUnary("not", ExprLiteral(LitInteger(4)))), ExprLiteral(LitInteger(5))))))
StmtIf(ExprLiteral(LitInteger(1)), [StmtExpr(ExprLiteral(LitInteger(2)))], [])
StmtIf(ExprLiteral(LitInteger(3)), [StmtExpr(ExprLiteral(LitInteger(4)))], [StmtExpr(ExprLiteral(LitInteger(5)))])
StmtIf(ExprLiteral(LitInteger(6)), [StmtExpr(ExprLiteral(LitInteger(7)))], [StmtIf(ExprLiteral(LitInteger(8)), [StmtExpr(ExprLiteral(LitInteger(9)))], [])])
StmtIf(ExprLiteral(LitInteger(10)), [StmtExpr(ExprLiteral(LitInteger(11)))], [StmtIf(ExprLiteral(LitInteger(12)), [StmtExpr(ExprLiteral(LitInteger(13)))], [StmtIf(ExprLiteral(LitInteger(14)), [StmtIf(ExprLiteral(LitInteger(16)), [StmtExpr(ExprLiteral(LitInteger(17)))], [])], [StmtExpr(ExprLiteral(LitInteger(18)))])])])
StmtExpr(ExprDict {})
StmtExpr(ExprDict { (ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))) })
StmtExpr(ExprDict { (ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))) })
StmtExpr(ExprDict { (ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))), (ExprBinary("add", ExprLiteral(LitInteger(3)), ExprLiteral(LitInteger(4))), ExprBinary("add", ExprLiteral(LitInteger(5)), ExprLiteral(LitInteger(6)))) })
StmtExpr(ExprDict { (ExprLiteral(LitInteger(0)), ExprLiteral(LitStr("0"))), (ExprLiteral(LitInteger(1)), ExprLiteral(LitStr("1"))), (ExprLiteral(LitInteger(2)), ExprLiteral(LitStr("2"))) })
StmtYield(Some(ExprTuple(ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)))))
StmtYield(None)
End
